File: Chapter 04/Python/section_4_2_downward_dependency/after/sql_order_repository.py
from .order_repository import OrderRepository
from .order import Order

class SqlOrderRepository(OrderRepository):
    """
    DATA ACCESS LAYER (LOW-LEVEL DETAIL).
    ARCHITECTURE NOTE: This is a concrete implementation. 
    It 'plugs into' the architecture by fulfilling the 
    OrderRepository contract.
    """
    def save(self, order: Order):
        print("(After Refactor) Saving order to SQL...")

File: Chapter 04/JS/section_4_2_downward_dependency/after/sqlOrderRepository.js
/**
 * DATA ACCESS LAYER.
 * ARCHITECTURE NOTE: In JS, this class implicitly satisfies the 
 * "Repository Contract" by implementing the save() method.
 */
class SqlOrderRepository {
    save(order) {
        console.log("(After Refactor) Saving order to SQL...");
    }
}

module.exports = SqlOrderRepository;